Bmx.Core.Test/EventTests.cs

```diff
@@ -0,0 +1,218 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Bmx.Core.State;
+using Bmx.Core.Test.State;
+using Moq;
+using NUnit.Framework;
+
+namespace Bmx.Core.Test {
+	[TestFixture]
+	[Category( "Events" )]
+	public class EventTests {
+		private Mock<IIdentityProvider<IAuthenticateState, IAuthenticatedState, IAccountState>>
+			_mockIdentityProvider;
+
+		private Mock<ICloudProvider<IRoleState>> _mockCloudProvider;
+
+		private IBmxCore<IAuthenticateState, IAuthenticatedState, IAccountState, IRoleState> _bmxCore;
+
+		private const string MockIdentityProviderName = "mockProvider";
+
+		[SetUp]
+		public void Setup() {
+			_mockIdentityProvider =
+				new Mock<IIdentityProvider<IAuthenticateState, IAuthenticatedState, IAccountState>>();
+			_mockCloudProvider = new Mock<ICloudProvider<IRoleState>>();
+
+			_bmxCore = new BmxCore<IAuthenticateState, IAuthenticatedState, IAccountState, IRoleState>(
+				_mockIdentityProvider.Object, _mockCloudProvider.Object );
+
+			_bmxCore.PromptUserName += provider => "";
+			_bmxCore.PromptUserPassword += provider => "";
+			_bmxCore.PromptMfaType += options => 0;
+			_bmxCore.PromptMfaInput += prompt => "";
+			_bmxCore.PromptAccountSelection += accounts => 0;
+			_bmxCore.PromptRoleSelection += roles => 0;
```

Comment:
By default if nothing is subscribed to an event C# will not complain.

I found this problematic WRT Bmx.Core, since any clients (such as Bmx.CommandLine, needs to subscribe for Bmx.Core to work as expected).

So I enforced that events are subscribed to using assertions (https://github.com/Brightspace/bmx/blob/dotnet/Bmx.Core/BmxCore.cs#L39). 

What this means however is the unit tests need to subscribe at least once before invoking Bmx.Core at all, else the assertions will fail.